### ISSUE TITLE
Tests written in Swift Testing framework are not running

### ISSUE THREAD

**Issue by NikolaGrujic91** [fastlane/fastlane#22088]
### New Issue Checklist

- [x ] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Issue Description

After migrating tests from XCTest to the new Swift Testing framework, migrated tests are not longer executed.
Using "--enable-experimental-swift-testing" did not help.

##### Command executed
    run_tests(workspace: ENV["WORKSPACE"],
              devices: [ENV["DEVICE_IPHONE"]],
              derived_data_path: ENV["DERIVED_DATA_PATH"],
              code_coverage: ENV["CODE_COVERAGE"],
              output_directory: "#{ENV["TESTS_OUTPUT_DIR"]}/#{ENV["TESTS_DIR_NAME"]}UtilityExtensions",
              scheme: "UtilityExtensions")

##### Complete output when running fastlane, including the stack trace and command used
There is no error, only the XCTests are running.

### Environment

<details><summary>🚫 fastlane environment 🚫</summary>

### Stack

| Key                         | Value                                                   |
| --------------------------- | ------------------------------------------------------- |
| OS                          | 14.5                                                    |
| Ruby                        | 3.3.0                                                   |
| Bundler?                    | false                                                   |
| Git                         | git version 2.39.4 (Apple Git-150.1)                    |
| Installation Source         | /usr/local/Cellar/fastlane/2.220.0/libexec/bin/fastlane |
| Host                        | macOS 14.5 (23F79)                                      |
| Ruby Lib Dir                | /usr/local/Cellar/ruby/3.3.0/lib                        |
| OpenSSL Version             | OpenSSL 3.2.0 23 Nov 2023                               |
| Is contained                | false                                                   |
| Is homebrew                 | true                                                    |
| Is installed via Fabric.app | false                                                   |
| Xcode Path                  | /Applications/Xcode 16 Beta.app/Contents/Developer/     |
| Xcode Version               | 16.0                                                    |
| Swift Version               | 6.0                                                     |

### System Locale

| Error                       |
| --------------------------- |
| No Locale with UTF8 found 🚫 |

### fastlane files:

<details><summary>`./Fastfile`</summary>

```ruby
# This file contains the fastlane.tools configuration
# You can find the documentation at https://docs.fastlane.tools
#
# For a list of all available actions, check out
#
#     https://docs.fastlane.tools/actions
#
# For a list of all available plugins, check out
#
#     https://docs.fastlane.tools/plugins/available-plugins
#

# Uncomment the line if you want fastlane to automatically update itself
# update_fastlane

# Fastlane Constants
default_platform(:ios)

  desc "Run unit tests"
  desc "The lane to run by developers"
  desc "#### Example:"
  desc "```\nfastlane ios unit_tests\n```"
  lane :unit_tests do
    run_tests(workspace: ENV["WORKSPACE"],
              devices: [ENV["DEVICE_IPHONE"]],
              derived_data_path: ENV["DERIVED_DATA_PATH"],
              code_coverage: ENV["CODE_COVERAGE"],
              output_directory: "#{ENV["TESTS_OUTPUT_DIR"]}/#{ENV["TESTS_DIR_NAME"]}UtilityExtensions",
              scheme: "UtilityExtensions")
  end
end

```
</details>

<details><summary>`./Appfile`</summary>

```ruby
# app_identifier("[[APP_IDENTIFIER]]") # The bundle identifier of your app
# apple_id("[[APPLE_ID]]") # Your Apple Developer Portal username


# For more information about the Appfile, see:
#     https://docs.fastlane.tools/advanced/#appfile

```
</details>

### fastlane gems

| Gem      | Version | Update-Status |
| -------- | ------- | ------------- |
| fastlane | 2.220.0 | ✅ Up-To-Date  |

### Loaded fastlane plugins:

**No plugins Loaded**

<details><summary><b>Loaded gems</b></summary>

| Gem                             | Version      |
| ------------------------------- | ------------ |
| error_highlight                 | 0.6.0        |
| did_you_mean                    | 1.6.3        |
| syntax_suggest                  | 2.0.0        |
| public_suffix                   | 5.0.5        |
| addressable                     | 2.8.6        |
| artifactory                     | 3.0.17       |
| jmespath                        | 1.6.2        |
| aws-partitions                  | 1.907.0      |
| aws-eventstream                 | 1.3.0        |
| aws-sigv4                       | 1.8.0        |
| aws-sdk-core                    | 3.191.6      |
| aws-sdk-kms                     | 1.78.0       |
| aws-sdk-s3                      | 1.146.1      |
| babosa                          | 1.0.4        |
| bundler                         | 2.5.4        |
| rexml                           | 3.2.6        |
| nkf                             | 0.1.3        |
| base64                          | 0.2.0        |
| CFPropertyList                  | 3.0.7        |
| colored                         | 1.2          |
| highline                        | 2.0.3        |
| commander                       | 4.6.0        |
| dotenv                          | 2.8.1        |
| emoji_regex                     | 3.2.3        |
| excon                           | 0.110.0      |
| faraday-em_http                 | 1.0.0        |
| faraday-em_synchrony            | 1.0.0        |
| faraday-excon                   | 1.1.0        |
| faraday-httpclient              | 1.0.1        |
| multipart-post                  | 2.4.0        |
| faraday-multipart               | 1.0.4        |
| faraday-net_http                | 1.0.1        |
| faraday-net_http_persistent     | 1.2.0        |
| faraday-patron                  | 1.0.0        |
| faraday-rack                    | 1.0.0        |
| faraday-retry                   | 1.0.3        |
| ruby2_keywords                  | 0.0.5        |
| faraday                         | 1.10.3       |
| faraday_middleware              | 1.2.0        |
| domain_name                     | 0.6.20240107 |
| http-cookie                     | 1.0.5        |
| faraday-cookie_jar              | 0.0.7        |
| fastimage                       | 2.3.1        |
| gh_inspector                    | 1.1.3        |
| uber                            | 0.1.0        |
| declarative                     | 0.0.20       |
| trailblazer-option              | 0.1.2        |
| representable                   | 3.2.0        |
| retriable                       | 3.1.2        |
| mini_mime                       | 1.1.5        |
| jwt                             | 2.8.1        |
| multi_json                      | 1.15.0       |
| os                              | 1.1.4        |
| signet                          | 0.19.0       |
| googleauth                      | 1.8.1        |
| httpclient                      | 2.8.3        |
| google-apis-core                | 0.11.3       |
| google-apis-androidpublisher_v3 | 0.54.0       |
| google-apis-playcustomapp_v1    | 0.13.0       |
| google-cloud-env                | 1.6.0        |
| google-cloud-errors             | 1.4.0        |
| google-cloud-core               | 1.7.0        |
| google-apis-iamcredentials_v1   | 0.17.0       |
| google-apis-storage_v1          | 0.31.0       |
| rake                            | 13.2.0       |
| digest-crc                      | 0.6.5        |
| google-cloud-storage            | 1.47.0       |
| json                            | 2.7.1        |
| mini_magick                     | 4.12.0       |
| naturally                       | 2.2.1        |
| optparse                        | 0.4.0        |
| plist                           | 3.7.1        |
| rubyzip                         | 2.3.2        |
| security                        | 0.1.5        |
| simctl                          | 1.6.10       |
| terminal-notifier               | 2.0.0        |
| unicode-display_width           | 2.5.0        |
| terminal-table                  | 3.0.2        |
| tty-screen                      | 0.8.2        |
| tty-cursor                      | 0.7.1        |
| tty-spinner                     | 0.9.3        |
| word_wrap                       | 1.0.0        |
| atomos                          | 0.1.3        |
| claide                          | 1.1.0        |
| colored2                        | 3.1.2        |
| nanaimo                         | 0.3.0        |
| xcodeproj                       | 1.24.0       |
| rouge                           | 2.0.7        |
| xcpretty                        | 0.3.0        |
| xcpretty-travis-formatter       | 1.0.1        |
| set                             | 1.1.0        |
| forwardable                     | 1.3.3        |
| logger                          | 1.6.0        |
| pathname                        | 0.3.0        |
| shellwords                      | 0.2.0        |
| cgi                             | 0.4.1        |
| date                            | 3.3.4        |
| timeout                         | 0.4.1        |
| securerandom                    | 0.3.1        |
| uri                             | 0.13.0       |
| openssl                         | 3.2.0        |
| digest                          | 3.1.1        |
| ipaddr                          | 1.2.6        |
| resolv                          | 0.3.0        |
| time                            | 0.3.0        |
| stringio                        | 3.1.0        |
| open-uri                        | 0.4.1        |
| mutex_m                         | 0.2.0        |
| net-http                        | 0.4.0        |
| net-protocol                    | 0.2.2        |
| ostruct                         | 0.6.0        |
| english                         | 0.8.0        |
| erb                             | 4.0.3        |
| abbrev                          | 0.1.2        |
| tempfile                        | 0.2.1        |
| delegate                        | 0.3.1        |
| fileutils                       | 1.7.2        |
| tmpdir                          | 0.2.0        |
| singleton                       | 0.2.0        |
| open3                           | 0.2.1        |
| prettyprint                     | 0.2.0        |
| pp                              | 0.5.0        |
| find                            | 0.2.0        |
| yaml                            | 0.3.0        |
| psych                           | 5.1.2        |
</details>

*generated on:* **2024-06-15**
</details>

**Comment by khoogheem** [comment]
any update

**Comment by bret3d** [comment]
Upgrading to Swift Testing and we need this as well.

**Comment by Ericdowney** [comment]
Yep. Seeing this too. Any updates?

**Comment by muncman** [comment]
+1 Running into this, as well.

**Comment by khoogheem** [comment]
So on one of my projects I added in:
`parallel_testing: false` 

and now all the test run as expected on my CI machine..

**Comment by xinsight** [comment]
I added `parallel_testing: false` and i'm still only seeing the XCTestCase tests when running `scan` (`run_tests`).

Weirdly, if I get a failure from a test in Testing, then the tests fail, and I can see the test that failed, but I don't see the test output. (Is xcpretty ignoring the output from Testing?)

**Comment by khoogheem** [comment]
here is my standard scan that I use now and see Swift Testing + XCTest both working
One project only SwitTesting, one Project that still has a Mix
```
 scan(
      scheme: "MyScheme",
      clean: true,
      device: "iPhone 15 (17.5)",
      skip_detect_devices: true,
      force_quit_simulator: true,
      code_coverage: false,
      output_types: "html",
      fail_build: true,
      parallel_testing: false,
      slack_username: "UnitTests",
      slack_icon_url: "https://developer.apple.com/assets/elements/icons/swift-testing/swift-testing-96x96_2x.png",
      slack_default_payloads: ['lane', 'test_result', 'git_branch', 'git_author', 'last_git_commit', 'last_git_commit_hash']
      )
  ```

**Comment by NikolaGrujic91** [comment]
Tried adding ```parallel_testing: false``` on fastlane 2.225.0 and 2.226.0, it is still not showing up results from Testing

**Comment by bret3d** [comment]
Also do not see Swift Testing results after trying `parallel_testing: false` on 2.226.0

**Comment by xinsight** [comment]
It seems that the test report also doesn't include `Testing` tests. If i run just a test that uses `Testing`, i get:

```
+-------------------------+
|      Test Results       |
+--------------------+----+
| Number of tests    | 10 |
| Number of failures | 0  |
+--------------------+----+
```
And the `junit` report only contains `xctest` results:

```reports/unit_tests.junit 
<?xml version='1.0' encoding='UTF-8'?>
<testsuites name='MinddocDigaTests.xctest' tests='0' failures='0'/>
```

So, somehow we need to get `Testing` tests added as a "testsuite"...

**Comment by xinsight** [comment]
Ok, so there are a few issues here:

1. junit output missing if `Testing` is run serially (not in parallel)

This seems to be a known issue for freaking **years** and still has not been fixed. https://github.com/swiftlang/swift-package-manager/issues/4752

Apparenty, a workaround is to enable parallel testing but limit to a single worker.

```
run_tests(
...
xcodebuild_command: "env NSUnbufferedIO=YES xcodebuild -parallel-testing-worker-count 1 -parallel-testing-enabled YES",
...
)
```
 
I tried this (Xcode 16.0) and but didn't see much difference. (I also didn't spend a ton of time on this.) There also seem to be like 10 ways to enable/disable parallel testing.

2. `xcpretty` doesn't know how to parse the output from swift Testing. Apparently, `xcbeautify` is now the recommended way to parse xcodebuild output. Even swift Testing output! But you have to install it separately.

```
brew install xcbeautify
```

And then you call it with:

```
run_tests(
 ...
      xcodebuild_formatter: "xcbeautify",
...
)
```

**Comment by bret3d** [comment]
Confirmed. Installing `xcbeautify` made the tests show up, albeit all over the place instead of in each test suite. Did not even need to change the fastfile as it looks for `xcbeautify` first over `xcpretty`. Thanks for sharing.

**Comment by xinsight** [comment]
> the tests show up, albeit all over the place instead

Make sure to disable parallel testing in your Xcode test plan.

<img width="645" alt="Image" src="https://github.com/user-attachments/assets/8d40696e-7926-40fb-80f1-23595ded1516" />
